pkg/controlplane/controller/defaultservicecidr/default_servicecidr_controller.go

```diff
@@ -138,8 +138,26 @@ func (c *Controller) sync() error {
 	serviceCIDR, err := c.serviceCIDRLister.Get(DefaultServiceCIDRName)
 	// if exists
 	if err == nil {
-		c.syncStatus(serviceCIDR)
-		return nil
+		// single to dual stack upgrade
+		if len(c.cidrs) == 2 && len(serviceCIDR.Spec.CIDRs) == 1 && c.cidrs[0] == serviceCIDR.Spec.CIDRs[0] {
+			klog.Infof("Updating default ServiceCIDR with CIDRs: %v", c.cidrs)
+			serviceCIDR = &networkingapiv1.ServiceCIDR{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: DefaultServiceCIDRName,
+				},
```

Comment:
this will clear any other existing metadata like annotations/labels/finalizers/managedFields, etc... should we deepcopy and just modify spec.CIDRs instead?

```go
serviceCIDR = serviceCIDR.DeepCopy()
serviceCIDR.Spec.CIDRs = c.cidrs
_, err := c.client.NetworkingV1().ServiceCIDRs().Update(context.Background(), serviceCIDR, metav1.UpdateOptions{})
...
```

Reply:
it will retry later